src/App.js

```diff
@@ -1,13 +1,136 @@
-import React from 'react';
+import React from "react";
+import TodoForm from "./components/TodoComponents/TodoForm";
+import TodoList from "./components/TodoComponents/TodoList";
+import Search from "./components/SearchComponent/Search";
+import "./App.css";
+
+// Set Task In LocalStorage
+if (localStorage.getItem("tasks") === null) {
+  localStorage.setItem("tasks", JSON.stringify([]));
+}
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      tasks: this.getTasks(),
+      task: ""
+    };
+  }
+
+  getTasks = () => {
+    return JSON.parse(localStorage.getItem("tasks"));
+  };
+
+  onChange = e => {
+    this.setState({ task: e.target.value });
+  };
+
+  addTodo = () => {
+    if (this.state.task.length === 0) {
+      return;
+    }
+    const newTask = {
+      id: Date.now(),
+      task: this.state.task,
+      completed: false
+    };
+    const newStateTasks = this.state.tasks.concat(newTask);
+    this.setDataToLocalStorage(newStateTasks);
+    this.setState({ tasks: newStateTasks, task: "" });
+  };
+
+  setDataToLocalStorage = data => {
+    localStorage.setItem("tasks", JSON.stringify(data));
+  };
+
+  toggleCompleted = id => {
+    const { tasks } = this.state;
+
+    tasks.find(task => {
```

Comment:
You have warnings during compilation, referred to this line and line 83. 
Eslint complin that you are not returning a value at the end of an arrow function, you could want to fix this.